src/libp2p/connection/yamux.rs

```diff
@@ -1564,6 +1604,33 @@ impl<'a, T> ExtractOut<'a, T> {
                 }
 
                 Outgoing::Idle => {
+                    // Send a "GoAway" frame if demanded.
+                    if let OutgoingGoAway::Required(code) = self.yamux.outgoing_goaway {
+                        let mut header = arrayvec::ArrayVec::new();
+                        header.push(0);
```

Comment:
what `0` is representing here?

Reply:
It's a Yamux version number. See https://github.com/hashicorp/yamux/blob/master/spec.md#version-field

There are several snippets of code that generate a header, and it's true that it wouldn't be a bad idea to move them to the `header` module for example, but I also don't think it's a big deal.